### PR TITLE
PIM-3736: Fix wrong count of products in Variant group view

### PR DESCRIPTION
**Pull request by grena**
| Q                    | A
| -------------------- | ---
| Bug fix?             | Y
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | Y (Build 1897)
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y
| Fixed tickets        | PIM-3736
| DB schema updated?   | N
| Migration script?    | -
| Doc PR               | -

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Form/Handler/GroupHandler.php [hunk 1]
```diff
@@ -72,6 +72,11 @@ public function process($group)
             }
         }
 
+        if ($group->getType()->isVariant()) {
+            $products = $this->productRepository->findAllForVariantGroup($group);
+            $group->setProducts($products);
```
Comment (filipsalpe) [inline src/Pim/Bundle/EnrichBundle/Form/Handler/GroupHandler.php]: Not a fan of this approach. Imagine if the user checked 100 products, 1 duplicates axis so the form is not valid - now we reset everything?

Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Form/Handler/GroupHandler.php]: In fact yes, if one fails, nothing is taken in account. But in the UI, the 100 checkboxes stay checked, it allows user to remove those who fail.

What do you propose ?

File: features/variant-group/add_products.feature [hunk 2]
```diff
@@ -69,3 +69,21 @@ Feature: Add products to a variant group
...
+    And I am on the "SANDAL" variant group page
+    Then the grid should contain 4 elements
+    And I check the row "sandal-white-37"
+    And I check the row "sandal-white-38"
+    And I check the row "sandal-white-39"
+    And I press the "Save" button
+    Then I should see "Products: 3"
+    And I check the row "duplicate-sandal"
+    And I press the "Save" button
+    Then I should see "Products: 3"
```
Comment (filipsalpe) [inline features/variant-group/add_products.feature]: But the row "duplicate-sandal" should be checked

File: features/variant-group/add_products.feature [hunk 3]
```diff
@@ -69,3 +69,21 @@ Feature: Add products to a variant group
       | 3       | Julia Stark - Julia@example.com                                   | groups   |                 |
       | 2       | Julia Stark - Julia@example.com (Comes from variant group SANDAL) | groups   | SANDAL          |
       | 1       | Admin Doe - admin@example.com                                     | sku      | sandal-white-37 |
+
+  Scenario: Reject product addition in a variant group, products count should be correct
```
Comment (nidup) [inline features/variant-group/add_products.feature]: you can ad the number of the jira as comment of this scenario

File: spec/Pim/Bundle/EnrichBundle/Form/Handler/GroupHandlerSpec.php [hunk 4]
```diff
@@ -87,15 +87,50 @@ function it_saves_a_variant_group_and_update_products_values_when_form_is_valid(
...
+
         $form->isValid()->willReturn(false);
+        $groupType->isVariant()->willReturn(false);
+        
+        $group->getType()->willReturn($groupType);
         $saver->save($group)->shouldNotBeCalled();
+        $this->process($group)->shouldReturn(false);
+    }
+
+    function it_returns_variant_group_products_if_submit_fails(
```
Comment (nidup) [inline spec/Pim/Bundle/EnrichBundle/Form/Handler/GroupHandlerSpec.php]: it_reloads_variant_group_products_if_submit_fails ?

File: features/variant-group/add_products.feature [hunk 5]
```diff
@@ -69,3 +69,23 @@ Feature: Add products to a variant group
...
+    Then the grid should contain 4 elements
+    And I check the row "sandal-white-37"
+    And I check the row "sandal-white-38"
+    And I check the row "sandal-white-39"
+    And I press the "Save" button
+    Then I should see "Products: 3"
+    And I check the row "duplicate-sandal"
+    And I press the "Save" button
+    Then I should see "Products: 3"
+    But the row "duplicate-sandal" should be checked
```
Comment (filipsalpe) [inline features/variant-group/add_products.feature]: Does it pass?

Comment (grena) [inline features/variant-group/add_products.feature]: Yep it does !

**Pull request merged**